bgpd/bgp_nht.c

```diff
@@ -1080,12 +1080,19 @@ static void sendmsg_zebra_rnh(struct bgp_nexthop_cache *bnc, int command)
  */
 static void register_zebra_rnh(struct bgp_nexthop_cache *bnc)
 {
+	struct interface *ifp;
+
 	/* Check if we have already registered */
 	if (bnc->flags & BGP_NEXTHOP_REGISTERED)
 		return;
 
 	if (bnc->ifindex) {
 		SET_FLAG(bnc->flags, BGP_NEXTHOP_REGISTERED);
+		ifp = if_lookup_by_index(bnc->ifindex, bnc->bgp->vrf_id);
+		if (ifp && if_is_up(ifp)) {
```

Comment:
Shouldn't be `if_is_operative()`?

Reply:
Yes, this fits better. this said, I see many functions using if_up(), and I think there may be some trouble in those calls that should be handled in a separate pr